src/lib/leaflet-directive.js

```diff
@@ -0,0 +1,32 @@
+import layerFactory from './leaflet-utils/layer-factory'
+import mapFactory from './leaflet-utils/map-factory'
+
+let map
+let currentLayers = []
+
+export default {
+  bind (el, { value }) {
+    const { config } = value
+    map = mapFactory(el, config)
+  },
+  update (_, { value, oldValue }) {
```

Comment:
Please see vue documentation for a more elegant solution:
https://vuejs.org/v2/guide/components-edge-cases.html#Dependency-Injection

as discussed here also:
https://github.com/KoRiGaN/Vue2Leaflet/issues/153